### PR TITLE
Fixed #28947 -- Fixed crash when coercing a translatable URL pattern to str.

### PR DESCRIPTION
**Pull request by tiltec**
Refs https://code.djangoproject.com/ticket/28947

I'm unsure where to put regression tests for those, any recommendations?

### CODE REVIEW

**Comment by charettes** [comment]
I'd say `tests/i18n/patterns/tests.py` and `tests/urlpatterns/tests.py` based on df41b5a05d4e00e80e73afe629072e37873e767a.

**Comment by tiltec** [comment]
Added a little test!

File: tests/i18n/patterns/tests.py [hunk 1]
```diff
@@ -163,6 +163,17 @@ def test_translate_url_utility(self):
             self.assertEqual(translate_url('/nl/gebruikers/', 'en'), '/en/users/')
             self.assertEqual(translation.get_language(), 'nl')
 
+    def test_string_representation(self):
+        from .urls.default import urlpatterns
```
Comment (charettes) [inline tests/i18n/patterns/tests.py]: Move this import to the top.

**Comment by tiltec** [comment]
@charettes 
I'm done for now. Are further changes needed?

**Comment by tiltec** [comment]
Done, thanks for reviewing!

**Comment by timgraham** [comment]
How about this for simpler tests:
```diff
diff --git a/tests/urlpatterns/tests.py b/tests/urlpatterns/tests.py
index b200aed..42ab82d 100644
--- a/tests/urlpatterns/tests.py
+++ b/tests/urlpatterns/tests.py
@@ -4,11 +4,25 @@ from django.core.exceptions import ImproperlyConfigured
 from django.test import SimpleTestCase
 from django.test.utils import override_settings
 from django.urls import Resolver404, path, resolve, reverse
+from django.urls.resolvers import RegexPattern, RoutePattern
+from django.utils.translation import gettext_lazy as _
 
 from .converters import DynamicConverter
 from .views import empty_view
 
 
+class RegexPatternTests(SimpleTestCase):
+
+    def test_str(self):
+        self.assertEqual(str(RegexPattern(_('^translated/$'))), '^translated/$')
+
+
+class RoutePatternTests(SimpleTestCase):
+
+    def test_str(self):
+        self.assertEqual(str(RoutePattern(_('translated/'))), 'translated/')
+
+
 @override_settings(ROOT_URLCONF='urlpatterns.path_urls')
 class SimplifiedURLTests(SimpleTestCase):
```

**Comment by tiltec** [comment]
@timgraham 
Makes sense, will rewrite the tests accordingly.

**Pull request merged**